### PR TITLE
Dev/mykhailo androsiuk

### PR DESCRIPTION
**Pull request by Mishytca**
Added tests are focused on the following issue:
https://github.com/xen-troops/moulin/issues/91

----

Added tests are focused on the following issue:
https://github.com/xen-troops/moulin/pull/94

----

To introduce integration tests we need to apply the change to the
moulin tests folder structure.

This patch moves unit tests into a separate 'unittest' folder.

### CODE REVIEW

**Review by lorc — CHANGES_REQUESTED** [review]
Comments for the second patch are the same as for the first.

Also, it is unclear for which issue the first test is written . I reverted changes made by #92 and the test still passes. Could you please explain how I can verify it?

Also-also, commit description refers to "issue" in two meanings: the issue with moulin code (when it failed when should work) and user error. So it is unclear what "the issue mentioned above
is detected during its execution" exactly mean. Does the test should detect the issue in the mouling code? If yes, why it passes when the issue is fixed?

File: tests/integration_tests/yocto_bulder/test_different_revisions_case/resources/test_different_revisions_case.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+desc: "Xen-Troops development setup for Renesas RCAR Gen3 hardware"
```
Comment (lorc) [inline tests/integration_tests/yocto_bulder/test_different_revisions_case/resources/test_different_revisions_case.yaml:1]: Nope :)

File: tests/integration_tests/yocto_bulder/test_different_revisions_case/resources/test_different_revisions_case.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        - "image_2_for_doma"
+      lunch_target: "target_for_doma"
+
+  domu:
+    build-dir: "%{YOCTOS_WORK_DIR}"
+    sources:
+      - type: git
+        url: "https://github.com/xen-troops/meta-example.git"
+        rev: BRANCH
+        dir: "dir1"
```
[the remark this answers is not in the document]
Comment (Mishytca) [reply tests/integration_tests/yocto_bulder/test_different_revisions_case/resources/test_different_revisions_case.yaml:30]: Hi @lorc , done.

File: tests/integration_tests/yocto_bulder/test_different_revisions_case/resources/test_different_revisions_case.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    builder:
+      type: yocto
+      build_target: core-image-weston
+      target_images:
+        - "image_1_for_domu"
+        - "image_2_for_domu"
+      lunch_target: "target_for_domu"
+
```
[the remark this answers is not in the document]
Comment (Mishytca) [reply tests/integration_tests/yocto_bulder/test_different_revisions_case/resources/test_different_revisions_case.yaml]: Hi @lorc , done.

File: tests/integration_tests/yocto_bulder/test_different_revisions_case/test_different_revisions_case.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import pytest
+import shutil
+import os
+
+
+@pytest.mark.integration
+def test_different_revisions():
+    script_path = os.path.abspath(__file__)
+    script_dir_path = os.path.dirname(script_path)
+    tmp_dir = os.path.join(script_dir_path, "tmp")
```
[the remark this answers is not in the document]
Comment (Mishytca) [reply tests/integration_tests/yocto_bulder/test_different_revisions_case/test_different_revisions_case.py]: Hi @lorc , done.

File: tests/integration_tests/yocto_bulder/test_different_revisions_case/resources/test_different_revisions_case.yaml (hunk 1 above)
Comment (Mishytca) [reply tests/integration_tests/yocto_bulder/test_different_revisions_case/resources/test_different_revisions_case.yaml]: Hi @lorc , done.

File: tests/integration_tests/yocto_bulder/test_different_revisions_case/test_different_revisions_case.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    try:
+        result = subprocess.run(["python", "../../../../../moulin.py", yaml_file],
+                                cwd=tmp_dir,
+                                stderr=subprocess.PIPE,
+                                text=True)
+
+        assert result.returncode != 0, ("The return code is not equal to '0'")
+        assert "YAMLProcessingError" in result.stderr, ("The expected error type is missing in the moulin output")
+        print(result.stderr)
```
[the remark this answers is not in the document]
Comment (Mishytca) [reply tests/integration_tests/yocto_bulder/test_different_revisions_case/test_different_revisions_case.py]: Hi @lorc , done.

File: tests/unittest/test_build_generator.py
[the remark this answers is not in the document]
Comment (Mishytca) [reply tests/unittest/test_build_generator.py:1]: Hi @lorc , done.

**Review by lorc — APPROVED** [review]
Great, thanks

`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Pull request merged**